### DISCUSSION TITLE
Add a fields order check

### DISCUSSION THREAD

**Discussion by aborruso** [frictionlessdata/frictionlessdata.io#674] [Ideas] [answered]
Hi,
for a lot of apps the fields order is not important.

This

| fieldA | fieldB | fieldC |
| --- | --- | --- |
| a | 1 | luke |
| b | 7 | sarah |
| a | 18 | ben |

and this

| fieldB | fieldA | fieldC |
| --- | --- | --- |
| 1 | a | luke |
| 7 | b | sarah |
| 18 | a | ben |

are the same, because the app look for simply `fieldA`, wherever it is.

I think it might be useful to have a resource check control, related to fields order. By default, without writing it, frictionless would give an error if the order was different from that of the scheme, but  if in the schema I wrote `fields-order:false`, I would have no errors, if the order was different from that of the diagram.

What do you think about?

**Comment by roll** [comment, marked as answer]
@aborruso 
Have you tried `detector.schema_sync=True` - https://framework.frictionlessdata.io/docs/guides/framework/detector-guide#schema-sync

I think it will do the trick.

**Reply by aborruso** [reply]
> Have you tried `detector.schema_sync=True` - https://framework.frictionlessdata.io/docs/guides/framework/detector-guide#schema-sync

I will try it, but I have a question for you.

In the doc I have "There is a way to sync provided schema based on a header row's field order". Then `detector.schema_sync=True` option is to sync a schema based on a field order.

I would like to be able to do a validation, in which the order of the fields of the schema can be different from that one of the input file.

**Reply by aborruso** [reply]
@roll how to apply `detector.schema_sync=True` via frictionless validate **cli**?

**Reply by roll** [reply]
@aborruso 
```bash
$ frictionless validate table.csv --schema schema-not-ordered.json --schema-sync
```

**Reply by aborruso** [reply]
> `frictionless validate table.csv --schema schema-not-ordered.json --schema-sync`

It works, great. Thank you

I have a note about it. If I read "Schema Sync" I have

> There is a way to sync provided schema based on a header row's field order. It's very useful when you have a schema that describes a subset or a superset of the resource's fields

Wouldn't it be better to have something like "if the order of the fields is not important for the validation, you can use `schema_sync` parameter and set it to true"?

**Reply by roll** [reply]
@aborruso 
I like your definition more than the current :smiley:
